xbmc/windows/GUIMediaWindow.cpp

```diff
@@ -1455,10 +1454,30 @@ void CGUIMediaWindow::OnInitWindow()
 
   m_rootDir.SetAllowThreads(true);
 
-  if (m_iSelectedItem > -1)
-    m_viewControl.SetSelectedItem(m_iSelectedItem);
-
   CGUIWindow::OnInitWindow();
+
+  m_viewControl.SetSelectedItem(GetMediaSelectedItem(m_vecItems->GetPath()));
+}
+
+int CGUIMediaWindow::GetMediaSelectedItem(const std::string& strDirectory)
+{
+  std::map<std::string, int>::iterator it;
+  it = m_mediaSelectedItems.find(strDirectory);
+  return (it != m_mediaSelectedItems.end()) ? it->second : 0;
+}
+
+void CGUIMediaWindow::SetMediaSelectedItem(const std::string& strDirectory, int iItem)
+{
+  std::map<std::string, int>::iterator it;
```

Comment:
you don't need all that for maps.
`m_mediaSelectedItems[strDirectory] = iItem;` will replace if it exists and it will add a new one if it doesn't